contracts/governance/extensions/GovernorProposalGuardian.sol

```diff
@@ -14,38 +14,31 @@ abstract contract GovernorProposalGuardian is Governor {
     event ProposalGuardianSet(address oldProposalGuardian, address newProposalGuardian);
 
     /**
-     * @dev Override {IGovernor-cancel} that implements the extended cancellation logic.
+     * @dev Override `_validateCancel` that implements the extended cancellation logic.
      *
      * * The {proposalGuardian} can cancel any proposal at any point in the lifecycle.
      * * if no proposal guardian is set, the {proposalProposer} can cancel their proposals at any point in the lifecycle.
-     * * if the proposal guardian is set, the {proposalProposer} keeps their default rights defined in {IGovernor-cancel} (calling `super`).
+     * * if the proposal guardian is set, the {proposalProposer} keeps their default rights defined in {IGovernor-_validateCancel} (calling `super`).
      */
-    function cancel(
-        address[] memory targets,
-        uint256[] memory values,
-        bytes[] memory calldatas,
-        bytes32 descriptionHash
-    ) public virtual override returns (uint256) {
-        address caller = _msgSender();
+    function _validateCancel(uint256 proposalId, address caller) internal view virtual override returns (bool) {
         address guardian = proposalGuardian();
 
         if (guardian == address(0)) {
             // if there is no proposal guardian
             // ... only the proposer can cancel
             // ... no restriction on when the proposer can cancel
-            uint256 proposalId = getProposalId(targets, values, calldatas, descriptionHash);
             address proposer = proposalProposer(proposalId);
-            if (caller != proposer) revert GovernorOnlyProposer(caller);
-            return _cancel(targets, values, calldatas, descriptionHash);
+            if (caller == proposer) return true;
         } else if (guardian == caller) {
             // if there is a proposal guardian, and the caller is the proposal guardian
             // ... just cancel
-            return _cancel(targets, values, calldatas, descriptionHash);
-        } else {
-            // if there is a proposal guardian, and the caller is not the proposal guardian
-            // ... apply default behavior
-            return super.cancel(targets, values, calldatas, descriptionHash);
+            return true;
         }
+
+        // if there is no guardian and the caller isn't the proposer or
+        // there is a proposal guardian, and the caller is not the proposal guardian
+        // ... apply default behavior
+        return super._validateCancel(proposalId, caller);
```

Comment:
```suggestion
        } else {
            // if there is no guardian and the caller isn't the proposer or
            // there is a proposal guardian, and the caller is not the proposal guardian
            // ... apply default behavior
            return super._validateCancel(proposalId, caller);
        }
```

Reply:
The approach here was to not require having two separate calls to super. If its required for readability we can do it this way.

Reply:
this is a consequence of this discussion: https://github.com/Amxx/openzeppelin-contracts/pull/10/files#r1925693063.

(let keep it to one thread)